src/main/java/igym/services/WorkoutService.java

```diff
@@ -92,4 +94,45 @@ public void deleteWorkout(UUID workoutId) {
         workoutRepository.save(workout);
         logger.info("Workout with id {} and its exercises have been inactivated", workoutId);
     }
+
+    /**
+     * Updates workout details.
+     * This method updates the workout's name and the exercise list.
+     * It does not update the gym associated with the workout.
+     */
+    @Transactional
+    public Workout updateWorkout(UUID workoutId, Workout updatedWorkout) {
+        logger.info("Attempting to update workout with id: {}", workoutId);
+        logger.debug("Update request with values: {}", updatedWorkout);
+
+        Workout existingWorkout = workoutRepository.findById(workoutId)
+                .orElseThrow(() -> {
+                    logger.warn("Workout with id {} not found", workoutId);
+                    return new WorkoutNotFoundException("Workout with id " + workoutId + " not found");
+                });
+
+        if (existingWorkout.getStatus() == Status.inactive) {
+            logger.warn("Workout with id {} is inactive and cannot be updated", workoutId);
+            throw new WorkoutNotFoundException("Workout with id " + workoutId + " not found");
+        }
```

Comment:
Hey, Daniel! I suggest to create a new private method in the Service layer called `findByIdAndStatus` that keeps all this logic. Then we can use it again when needed.

```suggestion
        Workout existingWorkout = findByIdAndStatus(workoutId)
```

Reply:
Nice, on the way